spec/Component/ProductListing/ListingPersisterSpec.php

```diff
@@ -0,0 +1,168 @@
+use BitBag\OpenMarketplace\Component\ProductListing\ListingPersister;
+use BitBag\OpenMarketplace\Component\ProductListing\ListingPersisterInterface;
+use BitBag\OpenMarketplace\Component\ProductListing\Repository\DraftImageRepositoryInterface;
+use BitBag\OpenMarketplace\Component\Vendor\Entity\VendorInterface;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\EntityManagerInterface;
+use Gaufrette\Filesystem;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\ImageInterface;
+use Sylius\Component\Core\Uploader\ImageUploaderInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+final class ListingPersisterSpec extends ObjectBehavior
+{
+    public function let(
+        FactoryInterface $productListingFactory,
+        DraftClonerInterface $draftCloner,
+        DraftGeneratorInterface $draftGenerator,
+        ImageUploaderInterface $imageUploader,
+        Filesystem $filesystem,
+        DraftImageRepositoryInterface $draftImageRepository,
+        EntityManagerInterface $entityManager,
+        DraftInterface $productDraft,
+        ImageInterface $image
+    ) {
+        $this->beConstructedWith(
+            $productListingFactory,
+            $draftCloner,
+            $draftGenerator,
+            $imageUploader,
+            $filesystem,
+            $draftImageRepository,
+            $entityManager
+        );
+
+        $productDraft->getImages()->willReturn(new ArrayCollection([$image->getWrappedObject()]));
+    }
+
+    public function it_is_initializable(): void
+    {
+        $this->shouldHaveType(ListingPersister::class);
+        $this->shouldImplement(ListingPersisterInterface::class);
+    }
+
+    public function it_creates_new_product_listing(
+        DraftInterface $productDraft,
+        VendorInterface $vendor,
+        ListingInterface $productListing,
+        FactoryInterface $productListingFactory
+    ): void {
+        $productListingFactory->createNew()->willReturn($productListing);
+        $productDraft->getCode()->willReturn('mug');
+
+        $this->createNewProductListing($productDraft, $vendor);
+
+        $productListing->setCode('mug')->shouldHaveBeenCalled();
+        $productListing->insertDraft($productDraft)->shouldHaveBeenCalled();
+        $productListing->setVendor($vendor)->shouldHaveBeenCalled();
+        $productDraft->setProductListing($productListing)->shouldHaveBeenCalled();
+        $productDraft->ownRelations()->shouldHaveBeenCalled();
+    }
+
+    public function it_resolves_latest_draft(
+        ListingInterface $productListing,
+        DraftGeneratorInterface $draftGenerator,
+        DraftInterface $productDraft
+    ): void {
+        $draftGenerator->generateNextDraft($productListing)->willReturn($productDraft);
+        $this->resolveLatestDraft($productListing);
+        $draftGenerator->generateNextDraft($productListing)->shouldBeCalledOnce();
+    }
+
+    public function it_updates_latest_draft_with(
```

Comment:
```suggestion
    public function it_updates_latest_draft(
```

With what? :thinking: